lib/ansible/executor/task_executor.py

```diff
@@ -553,10 +553,15 @@ def _execute(self, variables=None):
             if not self._task.evaluate_conditional(templar, variables):
                 display.debug("when evaluation is False, skipping this task")
                 return dict(changed=False, skipped=True, skip_reason='Conditional result was False', _ansible_no_log=self._play_context.no_log)
-        except AnsibleError:
-            # loop error takes precedence
+        except AnsibleError as e:
+            # Loop errors takes precendece but if both the loop and the conditional failed,
+            # include both information in the exception to prevent losing information useful
+            # for debugging.
             if self._loop_eval_error is not None:
-                raise self._loop_eval_error  # pylint: disable=raising-bad-type
+                raise type(self._loop_eval_error)(
+                    "Loop error was: '%s'. Conditional error was: '%s'." %
+                    (to_native(self._loop_eval_error), to_native(e))
+                )  # pylint: disable=raising-bad-type
```

Comment:
this might end up confusing users as to which error produced the failure, i suggest pushing the conditional error into a debug statement

Reply:
I initially had it in `-v` for that reason. Would `-v` be better as `ANSIBLE_DEBUG=1` is not that known among users?

Reply:
I think it would be more helpful to the user to show them both errors by default rather than hiding one in `-v`. Ideally we could have templating errors take precedence over loop errors and only display the templating error first, but that may not make sense in all situations.

If we don't want to display both as was done originally, then I'm fine with the current use of `-v` rather than using debug since debug is information overload for users.

Reply:
I agree it would be more helpful to display both errors but I was not able to come up with a clean way to do that with `Display` and/or exception messages. It felt like displaying both was cluttered and confusing. I am open to ideas though. :-)

I take it that this is at least an improvement over what we have now and putting this info into `-v` sounds like a "good UI"?

Reply:
Yeah, I had the same problem. I tried changing the exception type raised in `templar.template()` and catching that here, but still couldn't get it quite right.

Seems like putting it in `-v` is an improvement over what we have now until we can come up with something better.